delocate/delocating.py

```diff
@@ -59,17 +71,45 @@ def delocate_tree_libs(lib_dict, lib_path, root_path):
         Filtered `lib_dict` dict containing only the (key, value) pairs from
         `lib_dict` where the keys are the libraries copied to `lib_path``.
     """
-    copied_libs = {}
-    delocated_libs = set()
+    # Test for errors first to avoid getting half-way through changing the tree
+    libraries_to_copy, libraries_to_delocate = _analyze_tree_libs(
+        lib_dict, root_path
+    )
+    # Copy libraries and update lib_dict.
+    lib_dict, copied_libraries = _copy_required_libs(
+        lib_dict, lib_path, root_path, libraries_to_copy
+    )
+    # Update the install names of local and copied libaries.
+    _update_install_names(
+        lib_dict, root_path, libraries_to_delocate | copied_libraries
+    )
+    return libraries_to_copy
+
+
+def _analyze_tree_libs(
+    lib_dict,  # type: Mapping[Text, Mapping[Text, Text]]
+    root_path  # type: Text
+) -> Tuple[Dict[Text, Dict[Text, Text]], Set[Text]]:
```

Comment:
Nice split of the function - but I guess we are now depending on Python > 3.5.  That's fine, just noting.

Reply:
I did that by reflex.  I'll fix it since tests won't pass otherwise.